### ISSUE TITLE
Sleeping after creating dir?

### ISSUE THREAD

**Issue by ghost** [TGMDevelopment/JsonTGM#1]
https://github.com/TGMDevelopment/JsonTGM/blob/ee62bff35a4e09b14d4c4d0d32dcfd3556720838/src/main/java/ga/matthewtgm/json/files/JsonWriter.java#L25

Unless you can give me some proof that it's unable to wrote to the directory because it needs time to create, remove thread.sleep. it blocks the whole thread for a second

**Comment by Deftu** [comment]
It's required as the directory is still being created and if there are multiple parent directories then it can take up to a 10 milliseconds to create those included. The file cannot be rewritten to a null directory :P

**Comment by ghost** [comment]
> It's required as the directory is still being created and if there are multiple parent directories then it can take up to a 10 milliseconds to create those included. The file cannot be rewritten to a null directory :P

I have made a sample test, you really don't need the thread.sleep, it is literally slowing the code down unecessary lol try it for yourself
```java
public class Main {

	public static void main(String[] args) throws IOException {
		String d = "test/dir/cock/balls/ten/thorughsa/s/d/2e/23/2/3/2/32/e/a/d/a/w/d";
		long start = System.currentTimeMillis();
		File dir = new File(d);
		if(!dir.exists()) dir.mkdirs();
                File file = new File(d, "test.txt");
                if(!file.exists()) file.createNewFile();
		try(BufferedWriter bw = new BufferedWriter(new FileWriter(file))){
			bw.write("cock");
		}catch (Exception e) {}
		long end = System.currentTimeMillis();
		
		long time = (end-start);
		System.out.println(time + " ms");
		System.out.println(file.exists());
	}

}
```

**Comment by LowSpecCorgi** [comment]
> > It's required as the directory is still being created and if there are multiple parent directories then it can take up to a 10 milliseconds to create those included. The file cannot be rewritten to a null directory :P
> 
> I have made a sample test, you really don't need the thread.sleep lol try it for yourself
> 
> ```java
> public class Main {
> 
> 	public static void main(String[] args) throws IOException {
> 		String d = "test/dir/cock/balls/ten/thorughsa/s/d/2e/23/2/3/2/32/e/a/d/a/w/d";
> 		long start = System.currentTimeMillis();
> 		File dir = new File(d);
> 		if(!dir.exists()) dir.mkdirs();
>                 File file = new File(d, "test.txt");
>                 if(!file.exists()) file.createNewFile();
> 		try(BufferedWriter bw = new BufferedWriter(new FileWriter(file))){
> 			bw.write("cock");
> 		}catch (Exception e) {}
> 		long end = System.currentTimeMillis();
> 		
> 		long time = (end-start);
> 		System.out.println(time + " ms");
> 		System.out.println(file.exists());
> 	}
> 
> }
> ```

Not commenting on the issue, but I like your file directory

**Comment by Deftu** [comment]
Fixed in v0.2.9 [NOT RELEASED AS OF ISSUE BEING CLOSED]

**Comment by ghost** [comment]
> > It's required as the directory is still being created and if there are multiple parent directories then it can take up to a 10 milliseconds to create those included. The file cannot be rewritten to a null directory :P
> 
> I have made a sample test, you really don't need the thread.sleep, it is literally slowing the code down unecessary lol try it for yourself
> 
> ```java
> public class Main {
> 
> 	public static void main(String[] args) throws IOException {
> 		String d = "test/dir/cock/balls/ten/thorughsa/s/d/2e/23/2/3/2/32/e/a/d/a/w/d";
> 		long start = System.currentTimeMillis();
> 		File dir = new File(d);
> 		if(!dir.exists()) dir.mkdirs();
>                 File file = new File(d, "test.txt");
>                 if(!file.exists()) file.createNewFile();
> 		try(BufferedWriter bw = new BufferedWriter(new FileWriter(file))){
> 			bw.write("cock");
> 		}catch (Exception e) {}
> 		long end = System.currentTimeMillis();
> 		
> 		long time = (end-start);
> 		System.out.println(time + " ms");
> 		System.out.println(file.exists());
> 	}
> 
> }
> ```

nice test directory i like
